src/agent/src/rpc.rs

```diff
@@ -80,13 +80,32 @@ pub struct AgentService {
     sandbox: Arc<Mutex<Sandbox>>,
 }
 
+// A container ID must match this regex:
+//
+//     ^[a-zA-Z0-9][a-zA-Z0-9_.-]+$
+//
+fn verify_cid(id: &str) -> Result<()> {
+    let valid = id.len() > 1
+        && id.chars().next().unwrap().is_alphanumeric()
+        && id
+            .chars()
+            .all(|c| (c.is_alphanumeric() || ['.', '-', '_'].contains(&c)));
+
+    match valid {
+        true => Ok(()),
+        false => Err(anyhow!("invalid container ID: {:?}", id)),
+    }
+}
+
 impl AgentService {
     async fn do_create_container(
         &self,
         req: protocols::agent::CreateContainerRequest,
     ) -> Result<()> {
         let cid = req.container_id.clone();
 
+        let _ = verify_cid(&cid)?;
```

Comment:
Why not only verify it on the shim side, and there's no need to verify it again in agent?

Reply:
This PR also verifies the cid in the shim. But tbh, I think the agent should be as paranoid as possible.

That said, I don't like having to check twice, but we do need to check on the host side too since the runtime does env setup and we need to avoid cid's that might look like a path.

Reply:
Hi @jodh-intel 

Do we need check at https://github.com/kata-containers/kata-containers/blob/main/src/runtime/containerd-shim-v2/service.go#L360, since this is the entry of create an container instead of https://github.com/kata-containers/kata-containers/blob/main/src/runtime/containerd-shim-v2/service.go#L71, the "new" function is only for a shim, not for an container.

Reply:
Good catch @lifupan - branch updated and improved! :smile: